mantidproject/mantid
# Change representation of points from opaque cubes to translucent spheres

Description of work.

I replaced points in the splatter plot with the new [point gaussian representation](https://blog.kitware.com/paraview-5-introducing-the-point-gaussian-representation/).  This replaces the square pixel with a round sphere and allows us to modify both the size and opacity.

This change required modifying the underlying `vtkUnstructuredGrid` so that it stores point data instead of cell data. This also allowed me to simplify the loop generating the `vtkUnstructuredGrid` and eliminate several temporary containers.

**To test:**

code review: I suggest focusing on `vtkSplatterPlotFactory.cpp`, where the largest changes take place.

testing: Run the [IntegratePeaksMD](http://docs.mantidproject.org/nightly/algorithms/IntegratePeaksMD-v2.html) example and view TOPAZ_3132_md in the VSI. Verify that the bragg peaks are visible and "Point Gaussian" representation is selected. Check that the default size and opacity is reasonable. Open peaks in the VSI and click "show peaks in table". Ensure that the integrated peaks remain visible and all other points are hidden.



Fixes #15654.



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
